packages/browser/src/background.js

```diff
@@ -25,13 +24,13 @@ chrome.webRequest.onBeforeRequest.addListener(
     const hasUrl =
       url.pathname.match(/\/assets\/[a-zA-Z]+\./) &&
       !url.searchParams.has("inj") &&
-      (url.hostname.endsWith("discord.com") || url.hostname.endsWith("discordapp.com"));
-    if (hasUrl) blockedScripts.add(details.url);
+      (url.host.endsWith("discord.com") || url.host.endsWith("discordapp.com"));
 
-    if (blockedScripts.size === scriptUrls.length) {
-      const blockedScriptsCopy = Array.from(blockedScripts);
-      blockedScripts.clear();
+    const initScripts = ["web."];
```

Comment:
No allow in mv3??

Reply:
Because mv3 uses declarativeNetRequest we can't determine when to block or not at runtime (the logic is slightly different), that's the entire point of the DNR API. The block gets toggled on when the page loads and toggled off when `web.` loads. I adjusted the logic in onBeforeNavigate (see line 9) so that loading the popout won't re-enable the block.